_test/test_utilities/test_split_counts.m

```diff
@@ -0,0 +1,86 @@
+classdef test_split_counts < TestCase
+
+methods
+    function obj = test_split_counts(~)
+        obj@TestCase('test_split_counts');
+    end
+
+    function test_outputs_are_empty_if_inputs_are_empty(obj)
+        [chunks, idxs] = split_counts([], []);
+        assertTrue(isa(chunks, 'cell'))
+        assertTrue(isempty(chunks));
+        assertTrue(isa(idxs, 'double'))
+        assertTrue(isempty(idxs));
+    end
+
+    function test_outputs_have_length_1_if_max_counts_gt_total_counts(~)
+        counts = ones(1, 10);
+        max_counts = 11;
+        [chunks, idxs] = split_counts(counts, max_counts);
+        assertEqual(numel(chunks), 1);
+        assertEqual(chunks{1}, counts);
+        assertEqual(size(idxs), [2, 1]);
+        assertEqual(idxs, [1; numel(counts)]);
+    end
+
+    function test_outputs_have_length_1_if_max_counts_eq_total_counts(~)
+        counts = ones(1, 10);
+        max_counts = 11;
+        [chunks, idxs] = split_counts(counts, max_counts);
+        assertEqual(numel(chunks), 1);
+        assertEqual(chunks{1}, counts);
+        assertEqual(size(idxs), [2, 1]);
+        assertEqual(idxs, [1; numel(counts)]);
+    end
+
+    function test_error_if_counts_is_not_a_vector(~)
+        counts = ones(2, 10);
+        f = @() split_counts(counts, 100);
+        assertExceptionThrown(f, 'MATLAB:expectedVector');
+    end
+
+    function test_error_if_max_counts_is_not_a_scalar(~)
+        counts = ones(1, 10);
+        f = @() split_counts(counts, [1, 1]);
+        assertExceptionThrown(f, 'MATLAB:expectedScalar');
+    end
+
+    function test_error_if_max_counts_is_zero(~)
+        counts = ones(1, 10);
+        f = @() split_counts(counts, 0);
+        assertExceptionThrown(f, 'MATLAB:expectedPositive');
+    end
+
+    function test_chunking_correct_if_a_count_gt_max_counts(~)
```

Comment:
May seem silly, but what happens if counts is not an int? Error? Truncating if float?

Reply:
I don't see a reason why the algorithm shouldn't just work, whether `counts` are ints or not. I can add a test to this effect.

Perhaps the function should be renamed to something like `split_vector_on_sum` or just `split_vector`?

Reply:
I didn't mean counts, I had misunderstood what the function was doing. Ignore me.